Chandra/cmd_states/cmd_states.py

```diff
@@ -101,12 +102,33 @@ def decode_power(mnem):
 
 def _make_add_trans(transitions, date, exclude):
     def add_trans(date=date, **kwargs):
-        # if no key in kwargs is in the exclude set then update transition
+        # If no key in kwargs is in the exclude set then update transition
+        # And, when doing any update, update a bookkeeping key 'last_date'
```

Comment:
It's good.  Sometimes it's just not possible to write code that is self-evident and in this case long comments are required.  See:
https://github.com/astropy/astropy/blob/cc7c7860f08f5ff74d0e96436577ae7939a6b7d2/astropy/io/ascii/ui.py#L270
 
The guessing process is full of gotchas and subtleties...